web/src/main/java/com/marklogic/hub/web/service/FlowManagerService.java

```diff
@@ -137,29 +138,36 @@ public StepModel createStep(String flowName, Integer stepOrder, String stepId, S
         Flow flow = flowManager.getFlow(flowName);
         Step step = StepModel.transformToCoreStepModel(stepModel, stepJson);
 
-        if (step.getType() == null) {
+        if (step.getStepDefinitionType() == null) {
             throw new BadRequestException("Invalid Step Type");
         }
 
         // Only save step if step is of Custom type, for rest use the default steps.
-        switch (step.getType()) {
+        switch (step.getStepDefinitionType()) {
             case INGEST:
-                Step defaultIngest = getDefaultStepFromResources("hub-internal-artifacts/steps/ingest/marklogic/default-ingest.step.json", Step.StepType.INGEST);
+                StepDefinition defaultIngestDefinition = getDefaultStepDefinitionFromResources("hub-internal-artifacts/steps/ingest/marklogic/default-ingest.step.json", StepDefinition.StepType.INGEST);
```

Comment:
should be "INGESTION"

Reply:
Didn't want to break the current code.
Should be taken up as part of https://project.marklogic.com/jira/browse/DHFPROD-2096